apps/prairielearn/src/models/enrollment.sql

```diff
@@ -1,10 +1,26 @@
--- BLOCK insert_enrollment
-INSERT INTO
-  enrollments (user_id, course_instance_id)
-VALUES
-  ($user_id, $course_instance_id)
-RETURNING
-  *;
+-- BLOCK ensure_enrollment
+WITH
+  new_enrollment AS (
+    INSERT INTO
+      enrollments (user_id, course_instance_id)
+    VALUES
+      ($user_id, $course_instance_id)
+    ON CONFLICT DO NOTHING
+    RETURNING
+      *
+  )
+SELECT
+  *
+FROM
+  new_enrollment
+UNION ALL
+SELECT
+  *
+FROM
+  enrollments
+WHERE
+  user_id = $user_id
+  AND course_instance_id = $course_instance_id;
```

Comment:
ChatGPT recommended this - is there a better way to ensure that a query with `ON CONFLICT DO NOTHING` always returns a row?

Reply:
This is the way to do it. The alternative is to do `ON CONFLICT DO UPDATE SET user_id = user_id` or something like that, which will then always return a row. But this is horrible because it produces enormous write churn in the DB.